src/utarray.h

```diff
@@ -78,7 +84,9 @@ typedef struct {
 
 #define utarray_new(a,_icd) do {                                              \
   (a) = (UT_array*)malloc(sizeof(UT_array));                                  \
-  if ((a) == NULL) oom();                                                     \
+  if ((a) == NULL) {                                                          \
+    utarray_oom();                                                            \
```

Comment:
How about adding a `break;` after this line? Then we don't need to adjust the documentation and I can't imagine anything we would lose.

Reply:
Adding an `else` would be much clearer than adding a `break`; but I hesitate to do either one, because I'm really not trying to encourage people to write non-fatal OOM handlers. Supporting "non-fatal OOM" in uthash.h was a massive undertaking for (AFAIK) very little benefit. I'd rather people just avoid using `malloc`-that-can-fail in C to begin with.

Reply:
OK. But what would be your suggestion for shared libraries? They are not supposed to be fatal for the whole program at any time. I think your point is fair, and the default already supports your view. It's just that shared libraries have no choice and utarray offers a solution for them.

Reply:
I think the easiest thing to do is "ignore the problem; if you're using a shared library you are probably on Linux (where [malloc never fails](https://stackoverflow.com/questions/16674370/why-does-malloc-or-new-never-return-null)) or some other hosted desktop system (where malloc never fails AFAIK)."

The most correct thing to do would be to substitute in your own `malloc` which allocates out of a fixed-size memory resource created at the start of the subroutine that wants to use uthash/utarray. Then, on malloc failure (i.e. inside `utarray_oom()`), you `longjmp` back to the beginning of the subroutine, ["wink out"](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2015/p0089r0.pdf) your memory resource (so that it doesn't matter whether utarray leaks memory on failure), and continue via your fallback path.

Reply:
I understand your concerns now and I agree with it. To achieve the goal, instead of plainly leaving out `break;`, how about adding one line that omits a warning message, so as to remind users that they should either write a fatal OOM or do whatever they want to skip the initialization line? I'll also update the document for this.

Reply:
Just a friendly reminder: Your link also says that, on Linux, `overcommit_memory` can be turned off to make `malloc` failable again. Therefore, I can't agree with your view with respect to "`malloc` never fails"